fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1688,12 +1732,35 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 			writeThread = createGenericThreadPool();
 			readThreads = createGenericThreadPool();
 		}
-		writeThread->addThread(
-		    new Writer(db, defaultFdbCF, id, readIterPool, perfContextMetrics, SERVER_KNOBS->ROCKSDB_READ_PARALLELISM),
-		    "fdb-rocksdb-wr");
+		struct Writer* writer =
+		    new Writer(db, defaultFdbCF, id, readIterPool, perfContextMetrics, SERVER_KNOBS->ROCKSDB_READ_PARALLELISM);
+		if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
+			actors.push_back(updateHistogram(this, writer->metricPromiseStream->getFuture()));
+		}
+		writeThread->addThread(writer, "fdb-rocksdb-wr");
 		TraceEvent("RocksDBReadThreads").detail("KnobRocksDBReadParallelism", SERVER_KNOBS->ROCKSDB_READ_PARALLELISM);
 		for (unsigned i = 0; i < SERVER_KNOBS->ROCKSDB_READ_PARALLELISM; ++i) {
-			readThreads->addThread(new Reader(db, defaultFdbCF, readIterPool, perfContextMetrics, i), "fdb-rocksdb-re");
+			struct Reader* reader = new Reader(db, defaultFdbCF, readIterPool, perfContextMetrics, i);
+			if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
+				actors.push_back(updateHistogram(this, reader->metricPromiseStream->getFuture()));
+			}
+			readThreads->addThread(reader, "fdb-rocksdb-re");
+		}
+	}
+
+	ACTOR Future<Void> updateHistogram(RocksDBKeyValueStore* kvStore,
+	                                   FutureStream<std::tuple<int, const StringRef, double>> metricFutureStream) {
+		loop {
+			choose {
+				when(std::tuple<int, const StringRef, double> measure = waitNext(metricFutureStream)) {
+					const StringRef metricName = std::get<1>(measure);
+					double latency = std::get<2>(measure);
+					Histogram::getHistogram(ROCKSDBSTORAGE_HISTOGRAM_GROUP_ZHE,
```

Comment:
Please use a different group name.

Reply:
It just for testing usage. I will change it to the original name when I clean up the code.